bcipy/signal/model/offline_analysis.py

```diff
@@ -297,161 +297,167 @@ def analyze_gaze(
+            continue
 
         # Fit the model based on model type.
-        # Model 1: Fit Gaussian mixture (comp=2) on each symbol and each eye separately
         if model_type == "Individual":
-            model.fit(train_re)
-
-            means, covs = model.evaluate(test_re)
-
-            # Visualize the results:
-            visualize_gaze_inquiries(
-                le, re,
-                means, covs,
-                save_path=save_figures,
-                img_path=f'{data_folder}/{MATRIX_IMAGE_FILENAME}',
-                show=show_figures,
-                raw_plot=plot_points,
-            )
-            left_eye_all.append(le)
-            right_eye_all.append(re)
-            means_all.append(means)
-            covs_all.append(covs)
-
-            # TODO: Calculate scores for the test set.
-            # scores, predictions = model.predict(test_re)
-
-        # Model 2: Fit Gaussian mixture (comp=1) on a centralized data
+            # Model 1: Fit Gaussian mixture on each symbol separately
+            reshaped_data = preprocessed_data[sym].reshape((preprocessed_data[sym].shape[0]*preprocessed_data[sym].shape[2], preprocessed_data[sym].shape[1]))
+            model.fit(reshaped_data)
+
         if model_type == "Centralized":
-            # Centralize the data using symbol positions:
+            # Centralize the data using symbol positions and fit a single Gaussian.
             # Load json file.
             with open(f"{BCIPY_ROOT}/parameters/symbol_positions.json", 'r') as params_file:
                 symbol_positions = json.load(params_file)
+            
             # Subtract the symbol positions from the data:
-            centralized_data_left.append(model.reshaper.centralize_all_data(train_le, symbol_positions[sym]))
-            centralized_data_right.append(model.reshaper.centralize_all_data(train_re, symbol_positions[sym]))
+            for j in range(len(preprocessed_data[sym])):
+                centralized_data[sym].append(model.centralize(preprocessed_data[sym][j], symbol_positions[sym]))
+        
+        if model_type == "GP_SampleAverage":
+            # Instead of centralizing, take the time average:
+            for j in range(len(preprocessed_data[sym])):
+                temp = np.mean(preprocessed_data[sym][j], axis=1)
+                time_average[sym].append(temp)
+                centralized_data[sym].append(model.substract_mean(preprocessed_data[sym][j], temp))  # Delta_t = X_t - mu
+            centralized_data[sym] = np.array(centralized_data[sym])
+            time_average[sym] = np.mean(np.array(time_average[sym]), axis=0)
 
     if model_type == "Individual":
-        # Takes in all means and covs from individual symbols, and
-        # calculates the likelihoods and predictions for each test point.
-        # Convert test_dict to a list of arrays:
         accuracy = 0
         acc_all_symbols = {}
         counter = 0
+
+    if model_type == "GP_SampleAverage":
```

Comment:
I'm fine just getting this merged for now, but in the near future this should be refactored to be more manageable (testable, extensible, etc).

These `if` statements for each model type could be either a separate class or function for each case and a way of dispatching to the appropriate handler. I'm happy to assist with this work.

Reply:
I agree with your suggestion! This was a quick workaround for me to quickly be able to change and test the models as I was developing them. The nested `if` loops might look very confusing to some and I agree that it is not the best script for testing. I will refactor this script later with a patch.